Source/Core/VideoCommon/XFStructs.cpp

```diff
@@ -300,3 +303,246 @@ void PreprocessIndexedXF(u32 val, int refarray)
   const size_t buf_size = size * sizeof(u32);
   Fifo::PushFifoAuxBuffer(new_data, buf_size);
 }
+
+std::pair<std::string, std::string> GetXFRegInfo(u32 address, u32 value)
+{
+// Macro to set the register name and make sure it was written correctly via compile time assertion
+#define RegName(reg) ((void)(reg), #reg)
+#define DescriptionlessReg(reg) std::make_pair(RegName(reg), "");
+
+  switch (address)
+  {
+  case XFMEM_ERROR:
+    return DescriptionlessReg(XFMEM_ERROR);
+  case XFMEM_DIAG:
+    return DescriptionlessReg(XFMEM_DIAG);
+  case XFMEM_STATE0:  // internal state 0
+    return std::make_pair(RegName(XFMEM_STATE0), "internal state 0");
+  case XFMEM_STATE1:  // internal state 1
+    return std::make_pair(RegName(XFMEM_STATE1), "internal state 1");
+  case XFMEM_CLOCK:
+    return DescriptionlessReg(XFMEM_CLOCK);
+  case XFMEM_SETGPMETRIC:
+    return DescriptionlessReg(XFMEM_SETGPMETRIC);
+
+  case XFMEM_CLIPDISABLE:
+    return std::make_pair(
+        RegName(XFMEM_CLIPDISABLE),
+        fmt::format("{} clipping detection, {} trivial rejection, {} cpoly clipping acceleration",
+                    (value & 1) ? "Enable" : "Disable", (value & 2) ? "enable" : "disable",
+                    (value & 3) ? "enable" : "disable"));
```

Comment:
These are backwards (if the bit is set, it's disabled).  This should also be a union (even though dolphin doesn't currently do anything with these values).